argusim/simulation_manager/sim.py

```diff
@@ -38,116 +57,280 @@ def __init__(self, trial_number, log_directory, config_path) -> None:
-        
+
         # Get control input
         control_input = self.control_input
 
         # Step through the simulation
-        self.state = rk4(self.state, control_input, self.params, current_time, dt)
+        self.true_state = rk4(self.true_state, 0 * control_input, self.params, current_time, dt)
 
         # Mask state through sensors
-        measurement = self.sensors(current_time, self.state)
-        
+        measurement = self.sensors(current_time, self.true_state)
+
+        true_ECI_R_body = R.from_quat([*self.true_state[7:10], self.true_state[6]])
+
+        # Sun Sensor update
+        got_sun = False
+        SUN_IN_VIEW = True  # TODO actually check if sun is in view
+        if SUN_IN_VIEW and (current_time >= self.last_sun_sensor_measurement_time + SUN_SENSOR_DT):
+            sigma_sunsensor = np.deg2rad(3)  # [rad]
+            # TODO simulate RTC and use its drifting time
+            true_sun_ray_ECI = self.true_state[self.Idx["X"]["SUN_POS"]]
+            true_sun_ray_ECI /= np.linalg.norm(true_sun_ray_ECI)
+            true_sun_ray_body = true_ECI_R_body.inv().as_matrix() @ true_sun_ray_ECI
+            sunSensor = SunSensor(sigma_sunsensor)
+            # TODO use measurement variable
+            measured_sun_ray_in_body = sunSensor.get_measurement(true_sun_ray_body)
+            self.attitude_ekf.sun_sensor_update(
+                measured_sun_ray_in_body, true_sun_ray_ECI, current_time, sigma_sunsensor
+            )
+            self.last_sun_sensor_measurement_time = current_time
+            self.logr.log_v(
+                "sun_sensor_measurement.bin",
+                [current_time - self.J2000_start_time] + measured_sun_ray_in_body.tolist(),
+                ["Time [s]"] + [f"{axis} [-]" for axis in "xyz"],
+            )
+            got_sun = True
+
+        got_B = False
+        if current_time >= self.last_magnetometer_measurement_time + MAGNETOMETER_DT:
+            # TODO get b field using estiamted position not true position
+            true_Bfield_ECI = self.true_state[self.Idx["X"]["MAG_FIELD"]]
+            true_Bfield_ECI /= np.linalg.norm(true_Bfield_ECI)
+            measured_Bfield_in_body = measurement[9:12]
+            self.attitude_ekf.Bfield_update(
+                measured_Bfield_in_body, true_Bfield_ECI, current_time, self.params.magnetometer_direction_noise_std
+            )
+            self.last_magnetometer_measurement_time = current_time
+            self.logr.log_v(
+                "magnetometer_measurement.bin",
+                [current_time - self.J2000_start_time] + measured_Bfield_in_body.tolist(),
+                ["Time [s]"] + [f"{axis} [T]" for axis in "xyz"],
+            )
+            got_B = True
+
+        if got_B and got_sun:
+            if not self.attitude_ekf.initialized:
+                attitude_estimate = self.attitude_ekf.triad(
+                    true_sun_ray_ECI, measured_sun_ray_in_body, true_Bfield_ECI, measured_Bfield_in_body
```

Comment:
Again, use estimated position, simulated RTC's time, and sun model to compute `true_sun_ray_ECI` and `true_Bfield_ECI` that are passed into the ekf